datastore/common/common.go

```diff
@@ -1,11 +1,118 @@
+// 2023-10-10T00:00:00Z.
+//
+// NOTE: if any errors are encountered, warnings will be printed and fallback values will be used.
+func initValidTimeRange() {
+	dynTime0 := strings.ToLower(Getenv("DYNAMICTIME", "true"))
+	dynamicTime = ((dynTime0 != "false") && (dynTime0 != "no") && (dynTime0 != "0"))
+
+	// define function that derives a valid int64 time range component from env.var
+	// 'name', with a default value of 'defaultVal', and with the option of
+	// expressing the value as an ISO-8601 datetime as a fallback.
+	getSecs := func(name string, defaultVal int64, allowIso8601 bool) int64 {
+		val0 := strings.ToLower(Getenv(name, fmt.Sprintf("%d", defaultVal)))
+		var val int64
+		var err error
+
+		// first attempt to extract directly from int64
+		if val, err = strconv.ParseInt(val0, 10, 64); err != nil {
+			if allowIso8601 {
+				// then attempt to extract from ISO-8601 form
+				var t time.Time
+				if t, err = iso8601ToTime(val0); err != nil {
+					log.Printf(
+						"WARNING: failed to parse %s as an ISO-8601 datetime of the form "+
+						"2023-10-10T00:00:00Z: %s; falling back to default secs: %d",
+						name, val0, defaultVal)
+					val = defaultVal
+				} else {
+					val = t.Unix()
+				}
+			} else {
+				log.Printf(
+					"WARNING: failed to parse %s as an int64: %s; falling back to default secs: %d",
+					name, val0, defaultVal)
+				val = defaultVal
+			}
+		}
+		return val
+	}
+
+	loTimeName := "LOTIME"
+	hiTimeName := "HITIME"
+	defaultLoTimeSecs := int64(86400)
+	defaultHiTimeSecs := int64(0)
+
+	if dynamicTime {
+		loTimeSecs = getSecs(loTimeName, defaultLoTimeSecs, false)
+		hiTimeSecs = getSecs(hiTimeName, defaultHiTimeSecs, false)
+	} else {
+		// NOTE: in this case the defaults make little sense as they define a negative interval,
+		// but are kept like this to keep the documentation (e.g. in README) simple.
+		// A negative interval will be caught and fixed below. In any case, it hardly makes any
+		// sense to use defaults at all in the static case (i.e. you are likely to always want to
+		// specify the valid time range explicitly to match your test setup).
+		loTimeSecs = getSecs(loTimeName, defaultLoTimeSecs, true)
+		hiTimeSecs = getSecs(hiTimeName, defaultHiTimeSecs, true)
+	}
+
+	if hiTimeSecs <= loTimeSecs {
+		log.Printf(
+			"WARNING: hiTimeSecs (%d) <= loTimeSecs (%d); setting hiTimeSecs to loTimeSecs + 1",
+			hiTimeSecs, loTimeSecs)
+		hiTimeSecs = loTimeSecs + 1
+	}
+}
+
+func init() { // automatically called once on program startup (on first import of this package)
+	initValidTimeRange()
+}
+
+// See https://www.pauladamsmith.com/blog/2011/05/go_time.html
+var iso8601layout = "2006-01-02t15:04:05z" // note lower case!
+
+// iso8601ToTime converts time string ts of the form YYYY-MM-DDThh:mm:ssZ to time.Time.
+// Returns (val, nil) upon success, otherwise (time.Time{}, error).
+func iso8601ToTime(ts string) (time.Time, error) {
+        tm, err := time.Parse(iso8601layout, strings.ToLower(ts))
```

Comment:
Does this add clarity instead of just using `time.Parse(iso8601layout, strings.ToLower(ts))` directly?

Reply:
you mean not handling any error?!

Reply:
You are handling it anyway when calling (line 54), right? Actually ignoring the error you give htere?

Reply:
Sorry, I guess I misunderstood your point. Yes, I see your point now. The reason for this seemingly superfluous wrapping is that I copy/pasted the code from another repo where Iso8601ToTime was used outside the `common` package (and thus required an uppercase initial character, btw). In that case it's nice to not having to expose the iso8601layout outside the `common` package. So I suggest we keep the function as it is for now so it will be easy to make it visible to other packages (by changing the first 'i' to 'I' in the function name).

Reply:
Another (slightly more academic?) argument for keeping the function is that we only have to update two lines of code (both in `common.go`) should we decide to rename the `iso8601layout` variable.

Reply:
And you also don't have to worry about whether you are converting to the correct case everywhere since that is also done only in the wrapper function.

Reply:
All of these are good arguments, as soon as the function is used more then once... ;-)